core/rpc/core/src/impl/utils.rs

```diff
@@ -3101,10 +2998,11 @@ pub(crate) fn check_same_enum_value(items: &[JsonItem]) -> InnerResult<()> {
 }
 
 pub(crate) fn dedup_json_items(items: Vec<JsonItem>) -> Vec<JsonItem> {
-    let mut items = items;
-    items.sort_unstable();
-    items.dedup();
-    items
+    let mut set = items
+        .iter()
+        .map(JsonItem::to_owned)
+        .collect::<HashSet<JsonItem>>();
+    items.into_iter().filter(|item| set.remove(item)).collect()
```

Comment:
suggest to change the fn parameter to mutable reference:

```
pub(crate) fn dedup_json_items(items: &mut Vec<JsonItem>) {
    let mut set = HashSet::new();
    items.retain(|i| set.insert(i.clone()));
}
```

the invoker may be changed to
```diff
        let mut payload = payload;
-        payload.from.items = utils::dedup_json_items(payload.from.items);
+        utils::dedup_json_items(&mut payload.from.items);
```